java/refactoring.java/src/org/netbeans/modules/refactoring/java/ui/UIUtilities.java

```diff
@@ -57,14 +57,22 @@
 
     // XXX: Remove "test_" when #211651 is fixed
     private static final String TEST_JAVA_MIME_TYPE = "test_text/x-java"; // NOI18N
+    private static final String RECORD = "RECORD";
     /**
      * Element.Kind values allowed to be used when calling ElementHandle.create
      *
      * @see javax.lang.model.element.ElementKind
      * @see org.netbeans.api.java.source.ElementHandle
      */
     public static EnumSet allowedElementKinds = EnumSet.of(ElementKind.PACKAGE, ElementKind.CLASS, ElementKind.INTERFACE, ElementKind.ENUM, ElementKind.ANNOTATION_TYPE, ElementKind.METHOD, ElementKind.CONSTRUCTOR, ElementKind.INSTANCE_INIT, ElementKind.STATIC_INIT, ElementKind.FIELD, ElementKind.ENUM_CONSTANT, ElementKind.TYPE_PARAMETER);
-
+    static {
+        ElementKind recKind = null;
+        try {
+            recKind = ElementKind.valueOf(RECORD);
```

Comment:
You are using `TreeShims.RECORD` in the `java/java.source.base/src/org/netbeans/api/java/source/TreeUtilities.java`. Can't be it used?

Reply:
Treshims is not currently supported in java.refactoring module and to support this TreeShimCopiler class need to be modified. So instead of adding dependency on that class for the TreeShims.ROCORD had created the constant locally.

Reply:
I see. I'm not an expert this area, so just asked. Thanks! Please add `// NOI18N` to `private static final String RECORD = "RECORD";`

Reply:
Ok, will add.

Reply:
There is some issue in this PR. Created another PR https://github.com/apache/netbeans/pull/2310  and addressed all the review comments. Please verify.